hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java

```diff
@@ -545,9 +554,17 @@ public void run() {
      * Called for an orderly shutdown.
      */
     public void shutdown() {
+        logger.info("Shutting down Hedera node");
         shutdownGrpcServer();
 
         if (daggerApp != null) {
+            logger.debug("Shutting down the state");
+            final var state = daggerApp.workingStateAccessor().getHederaState();
+            if (state instanceof MerkleHederaState mhs) {
+                mhs.close();
+            }
```

Comment:
Do we only want to close state if it's a `MerkleHederaState`? A `close()` method seems like the type of thing we'd want on any state as a last chance at cleanup, and classes that don't matter (e.g. test fixtures) could have a no-op. Maybe that's overkill though.

Reply:
I filed another issue to resolve lifecycle of the node. It should maybe be that platform initiates the communication with merkle nodes, in which case we just remove the close method workaround. We'll sort it out then.